integrations/langfuse/tests/test_tracer.py

```diff
@@ -400,3 +410,130 @@ def test_init_with_tracing_disabled(self, monkeypatch, caplog):
 
             LangfuseTracer(tracer=MockTracer(), name="Haystack", public=False)
             assert "tracing is disabled" in caplog.text
+
+    def test_context_cleanup_after_nested_failures(self):
```

Comment:
I recently moved this to test_tracing.py in #2203 and marked as an integration test (requiring API keys)
I'm also OK to moving back to this module and transforming it into a unit test. The important thing is not to have duplicates.